A40-csds-support.md

```diff
@@ -0,0 +1,328 @@
+gRPC receives them**.
+
+#### Detail: Cache Lifecycle
+
+gRPC doesn't use xDS messages directly, but interprets them into language-native
+class/struct. The lifecycle of the cached xDS messages should be identical to
+the interpreted structure in each stack, so there is no memory management logic
+change needed.
+
+#### Detail: Generated Node Information
+
+gRPC loads Node information from a [bootstrap
+file](https://github.com/grpc/proposal/blob/master/A27-xds-global-load-balancing.md#xdsclient-and-bootstrap-file),
+which includes the identification that will be used in the control plane. gRPC's
+xDS client also generates `user_agent_name`, `user_agent_version`, and
+`client_features` before sending the `Node` information to the control plane.
+This information can be very helpful to locate the culprit release and
+debugging. This doc recommends the implementation to provide the
+`ClientStatusResponse.ClientConfig.node` field with the `Node` information that
+the control plane will receive. Here is an example of a filled `Node`:
+
+```json
+{
+  "node": {
+    "id": "c591240a-b3b6-4761-aada-3fa43ec6a852",
+    "cluster": "cluster",
+    "metadata": {
+      "...": "..."
+    },
+    "locality": {
+      "zone": "zone"
+    },
+    "userAgentName": "gRPC Java",
+    "userAgentVersion": "1.35.1-SNAPSHOT",
+    "clientFeatures": ["envoy.lb.does_not_support_overprovisioning"]
+  }
+}
+```
+
+#### Detail: Node Matching
+
+The node matching mechanism is not designed for gRPC’s use case, but for
+querying the control plane. Technically, gRPC application only serves as one xDS
+node, so there is no need to differentiate with xDS client's status to reply.
+However, gRPC should reject the request with Node Matcher that failed to match
+with the Node information of the gRPC application. This feature has lower
+priority. To preserve forward compatibility, this doc proposes to throw an
+[INVALID_ARGUMENT](https://github.com/grpc/grpc/blob/master/doc/statuscodes.md)
+if the CSDS implementation observe an non-empty Node Matcher.
+
+
+## Alternatives
+
+### Solution: Config Dump By Attaching An HTTP Server
+
+Although it is the most straightforward way of implementing an admin interface,
+Java doesn’t have a good enough built-in web server and it is challenging to
+introduce yet another dependency into Core. Exposing application states via HTTP
+will be challenging from engineering perspective.
+
+### Solution: Config Dump via Channelz
+
+Though only a subset of xDS resources is applicable to a channel, xDS configs
+(listeners, routes, clusters, endpoints) are global resources to a gRPC
+application. One unsolved problem is that if we choose to use CSDS to expose xDS
+configs, **the users won’t have the ability to query the set of effective xDS
+configs for a channel**. The CSDS is not designed for finer granularity than
+individual processes.
+
+If we decided to support sharing XdsClient across multiple channels, the
```

Comment:
This isn't an "if"; we've already done this.

Reply:
Removed the if statement.